apps/dashboard/src/components/workflow-editor/workflow-canvas.tsx

```diff
@@ -275,9 +277,36 @@ const WorkflowCanvasChild = ({ steps, readOnly }: { steps: Step[]; readOnly?: bo
 };
 
 export const WorkflowCanvas = ({ steps, readOnly }: { steps: Step[]; readOnly?: boolean }) => {
+  const has = useHasPermission();
+  const isReadOnly = readOnly || !has({ permission: PermissionsEnum.WORKFLOW_CREATE });
+
   return (
     <ReactFlowProvider>
-      <WorkflowCanvasChild steps={steps || []} readOnly={readOnly} />
+      <div className="relative h-full w-full">
+        <WorkflowCanvasChild steps={steps || []} readOnly={readOnly} />
+
+        {isReadOnly && (
+          <>
+            <div
+              className="border-warning/20 pointer-events-none absolute inset-x-0 top-0 border-t-[0.5px]"
+              style={{
+                position: 'absolute',
+                height: '100%',
+                background: 'linear-gradient(to bottom, hsl(var(--warning) / 0.08), transparent 4%)',
+                transition: 'border 0.3s ease-in-out, background 0.3s ease-in-out',
+              }}
+            />
+            <div className="absolute left-4 top-4 z-50">
+              <InlineToast
+                className="bg-warning/10 border shadow-md"
+                variant={'warning'}
+                description="Nice! You can see this, but changes are locked down."
```

Comment:
```suggestion
                description="Content visible but locked for editing. Contact an admin for edit access."
```

Not sure the "Nice" part is adding value, wdyt about something like this?

Reply:
I agree, sounds better.